frontend/lib/cache.ts

```diff
@@ -1,14 +1,18 @@
 import LRU from 'lru-cache';
+import { logAsync } from '@ifixit/helpers';
+
 const cacheStore = new LRU({ max: 100 });
 
-export function cache<T>(key: string, fetch: () => T, ttl: number): T {
-   const fromCache = cacheStore.get(key) as T | undefined;
+export function cache<T>(
+   key: string,
+   fetch: () => Promise<T>,
+   ttl: number
+): Promise<T> {
+   const fromCache = cacheStore.get(key) as Promise<T> | undefined;
    if (fromCache !== undefined) {
-      console.log(`Got '${key}' from cache`);
       return fromCache;
    }
-   const result = fetch();
-   console.log(`Storing '${key}' in cache`);
+   const result = logAsync(`Cache miss for '${key}'`, () => fetch());
```

Comment:
I think this could be a bare `fetch`. Also can we use another name since fetch is overloaded?

Reply:
Fixed!